### PR TITLE
test: add submitGuess coverage to useGameActions

### LINKED ISSUE #117: [Testing] useGameActions missing submitGuess test coverage
## Problem

`src/hooks/actions/__tests__/useGameActions.test.ts` only covers `submitRange` (2 tests). The `submitGuess` function has zero test coverage.

## Impact

Can't catch:
- Validation failures not properly displayed
- Authentication edge case handling (lines 128-159)
- Optimistic update not rolling back on specific error types

## Location

`src/hooks/actions/useGameActions.ts` (submitGuess function)
`src/hooks/actions/__tests__/useGameActions.test.ts` (missing tests)

## Acceptance Criteria

- [ ] Add tests for `submitGuess` covering:
  - Successful submission
  - Validation failure handling
  - Auth state edge cases

### PR DESCRIPTION
**Pull request by phrazzld**
## Summary

Closes #117

`useGameActions.test.ts` had zero coverage for `submitGuess` — only `submitRange` was tested. This left three real risk categories untested: validation failures, auth state edge cases (authenticated but null userId), and error-path behavior (optimistic guess retained vs. rolled back).

## Changes

- `src/hooks/actions/__tests__/useGameActions.test.ts` — adds `describe("useGameActions - submitGuess")` block with 9 tests covering the full `submitGuess` surface area.

## Acceptance Criteria

- [x] Successful submission (authenticated): optimistic `addGuess` fires, mutation called, returns `true`
- [x] BC year (negative number): handled identically to AD years
- [x] Anonymous user: local `addGuess` fires, no server call, returns `true`
- [x] Validation failure — no puzzle loaded: skips `addGuess`, returns `false`
- [x] Validation failure — out-of-bounds year: skips `addGuess`, returns `false`
- [x] Validation failure — all 6 guesses exhausted: skips `addGuess`, returns `false`
- [x] Auth edge case — `isAuthenticated=true` but `userId=null`: optimistic `addGuess` fires (no rollback), returns `false`
- [x] Server failure — generic error: optimistic guess kept, returns `true`
- [x] Server failure — `ConvexIdValidationError`: optimistic guess kept, returns `true`

## Manual QA

```bash
pnpm test src/hooks/actions/__tests__/useGameActions.test.ts
```

Expected: all 11 tests pass (2 existing `submitRange` + 9 new `submitGuess`).

```bash
pnpm test && pnpm type-check && pnpm lint
```

Expected: clean across the board.

## Before / After

**Before:** `useGameActions.test.ts` contained 2 tests — both for `submitRange`. `submitGuess` was entirely untested. Bugs in validation, auth branching, or error handling would go undetected. The `submitGuess` test block had repeated `let returnValue / await act` boilerplate (8 instances) and no coverage for BC years — a core game feature.

**After:** 9 tests cover every branch of `submitGuess`: happy path (auth + anon), BC year handling, three validation gates, the authenticated-but-null-userId edge case, and both error-path behaviors. A `callSubmitGuess` helper eliminates test boilerplate. Filed #183 for the remaining `resetGame` coverage gap. No behavior changes — tests and refactoring only.

## Test Coverage

- **File:** `src/hooks/actions/__tests__/useGameActions.test.ts`
- **New describe block:** `useGameActions - submitGuess` (lines 108–246)
- **Gaps:** `resetGame` untested → tracked in #183. No integration-level test verifying the Convex mutation payload shape — those belong in E2E tests, not unit tests.

### CODE REVIEW

File: src/hooks/actions/__tests__/useGameActions.test.ts [hunk 1]
```diff
@@ -102,3 +103,143 @@ describe("useGameActions - submitRange", () => {
...
+
+describe("useGameActions - submitGuess", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+    mockSubmitGuessMutation.mockReset();
+    mockSubmitRangeMutation.mockReset();
+    vi.mocked(useMutationWithRetry)
+      .mockReturnValueOnce(mockSubmitGuessMutation)
+      .mockReturnValue(mockSubmitRangeMutation);
+  });
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply src/hooks/actions/__tests__/useGameActions.test.ts]: Fixed in fe1624d — added `vi.mocked(useMutationWithRetry).mockReturnValue(mockSubmitRangeMutation)` to the submitRange `beforeEach` so each suite resets the mock implementation independently, preventing order-dependent leakage.
